annotations/certmanager-annotations/src/main/java/io/dekorate/certmanager/manifest/CertificateResourceGenerator.java

```diff
@@ -56,27 +58,31 @@ public void generate(CertificateConfig config) {
     LOGGER.info("Processing certificate config.");
     validate(config);
 
+    String defaultName = configurationRegistry.get(BaseConfig.class).map(BaseConfig::getName)
+        .orElse(config.getProject().getBuildInfo().getName());
+    String name = defaultIfEmpty(config.getName(), defaultName);
```

Comment:
This is the important fix: Instead of getting the name from the generated deployment resource, it will get it from the BaseConfig.
I've confirmed that this works fine for the kubernetes and openshift extensions.